components/inputs/input-text.js

```diff
@@ -476,7 +476,10 @@ class InputText extends FocusMixin(LabelledMixin(FormElementMixin(SkeletonMixin(
 			label = this.getAttribute('aria-label');
 		}
 		if (label) {
-			const unitLabel = this.unit ? ` ${this.unit}` : '';
+			let unitLabel = '';
+			if (this.unit) {
+				unitLabel = this.unitLabel ? ` ${this.unitLabel}` : this.unit;
```

Comment:
Slight variation here -- `unit-label` has a space between the label and the unit (e.g. "3 pixels", "3 out of 5") vs. when just `unit` is present there is no space (e.g. "3%"). We may need to make this more flexible in the future.

Reply:
I wonder if we should make the `unit-label` required if `unit` is provided and it's not `%`. I can't really think of any other abbreviations that the browser/screen reader would automatically handle.

Reply:
I like that idea -- will update.

Reply:
I've added validation to ensure that `unit-label` is set whenever `unit` isn't `%`.

For the space vs. no-space, I think this was a bug actually. The existing code put a space when the label wasn't hidden, but didn't include the space for `label-hidden`. Because it was always just `label%` or `label %` it likely wasn't noticed. I've updated it to always use a space.